lib/streamlit/runtime/forward_msg_queue.py

```diff
@@ -81,14 +81,22 @@ def enqueue(self, msg: ForwardMsg) -> None:
         self._delta_index_map[delta_key] = len(self._queue)
         self._queue.append(msg)
 
-    def clear(self, retain_lifecycle_msgs: bool = False) -> None:
+    def clear(
+        self,
+        retain_lifecycle_msgs: bool = False,
+        fragment_ids_this_run: list[str] | None = None,
```

Comment:
I am a little bit worried of making the `forward_msg_queue` aware of fragment ids, but in order to avoid this we might need to replicate the queue-cleaning logic 🤔 so I opted for this instead.